packages/create-catalyst/src/commands/create.ts

```diff
@@ -1,37 +1,94 @@
 import { input, select } from '@inquirer/prompts';
 import chalk from 'chalk';
 import { exec as execCallback } from 'child_process';
+import { pathExistsSync } from 'fs-extra/esm';
+import kebabCase from 'lodash.kebabcase';
+import { join } from 'path';
 import { promisify } from 'util';
+import { z } from 'zod';
 
-import { type CreateCommandOptions } from '../index';
+import { type CreateCommandOptions } from '..';
+import { checkStorefrontLimit } from '../utils/check-storefront-limit';
 import { cloneCatalyst } from '../utils/clone-catalyst';
 import { Https } from '../utils/https';
 import { installDependencies } from '../utils/install-dependencies';
 import { login } from '../utils/login';
-import { projectConfig } from '../utils/project-config';
+import { parse } from '../utils/parse';
 import { spinner } from '../utils/spinner';
 import { writeEnv } from '../utils/write-env';
 
 const exec = promisify(execCallback);
 
 export const create = async (options: CreateCommandOptions) => {
-  const { packageManager } = options;
-  const bigCommerceApiUrl = `https://api.${options.bigcommerceHostname}`;
-  const bigCommerceAuthUrl = `https://login.${options.bigcommerceHostname}`;
-  const { projectName, projectDir } = await projectConfig({
-    projectDir: options.projectDir,
-    projectName: options.projectName,
-  });
-  const { storeHash, accessToken } = await login(
-    bigCommerceAuthUrl,
-    options.storeHash,
-    options.accessToken,
-  );
+  const { packageManager, ghRef } = options;
+
+  const URLSchema = z.string().url();
+  const sampleDataApiUrl = parse(options.sampleDataApiUrl, URLSchema);
+  const bigcommerceApiUrl = parse(`https://api.${options.bigcommerceHostname}`, URLSchema);
+  const bigcommerceAuthUrl = parse(`https://login.${options.bigcommerceHostname}`, URLSchema);
+
+  let projectName;
+  let projectDir;
+  let storeHash = options.storeHash;
+  let accessToken = options.accessToken;
+  let channelId;
+  let customerImpersonationToken = options.customerImpersonationToken;
```

Comment:
These values _may_ be parsed from options, but typically need additional validation, formatting, etc. which is denoted below by the `if (options.property)` statements below